app/components/form_components/schools_autocomplete/script.js

```diff
@@ -0,0 +1,67 @@
+import accessibleAutocomplete from 'accessible-autocomplete'
+
+const $form = document.querySelector('[data-module="app-schools-autocomplete"]')
+const idElement = document.getElementById('lead-school-id')
+
+let statusMessage = null
+
+const findSchools = (query, populateResults) => {
+  statusMessage = 'Loading...'
+
+  window.fetch(`/api/schools?query=${query}&lead_school=true`)
+    .then(response => response.json())
+    .then(data => {
+      if (data === undefined) {
+        return
+      }
+
+      const schools = data.schools
+
+      if (schools.length === 0) {
+        statusMessage = 'No results found'
+      }
+
+      return schools
+    })
+    .then(schools => populateResults(schools))
+    .catch(err => console.log(err))
+}
```

Comment:
```suggestion
const guard = data => {
  if (data === undefined) {
    return new Error('some thing really bad has happened')
  }
  return  data
}

const mapToSchools = data => data.schools.map(school => {name: school.name, id: school.id})

const tryUpdateStatusMessage = schools => {
  if (schools.length === 0) {
    statusMessage = 'No results found'
   }
   return  schools
}
      
const findSchools = (query, populateResults) => {
  statusMessage = 'Loading...' // Shared state

  window.fetch(`/api/schools?query=${query}&lead_school=true`)
    .then(response => response.json())
    .then(guard)
    .then(mapToSchools)
    .then(tryUpdateStatusMessage)
    .then(populateResults)
    .catch(console.log)
}
```


`mapToSchools` we may need to do that on serverside as it quite heavy